core/src/main/java/org/arquillian/smart/testing/spi/JavaSPILoader.java

```diff
@@ -0,0 +1,93 @@
+package org.arquillian.smart.testing.spi;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+
+public class JavaSPILoader {
+
+    public JavaSPILoader() {
+    }
+
+    /**
+     * Get all Java services that implements given interface.
+     * @param service interface
+     * @param <SERVICE>
+     * @return Iterable of all services implementing service interface and present in classpath.
+     */
+    public <SERVICE> Iterable<SERVICE> all(Class<SERVICE> service) {
+        return ServiceLoader.load(service);
+    }
+
+    /**
+     * Get all Java services that implements given interface and meets the given predicate.
+     * @param service interface
+     * @param predicate to set filtering options
+     * @param <SERVICE>
+     * @return Iterable of all services implementing service interface, meeting predicate condition and present in classpath.
+     */
+    public <SERVICE> Collection<SERVICE> all(Class<SERVICE> service, Predicate<SERVICE> predicate) {
+        return StreamSupport.stream(all(service).spliterator(), false).filter(predicate).collect(Collectors.toList());
```

Comment:
Why in some cases we are returning the `Collection` and in others `Iterable`? Should we make it unified?